lib/pure/asynchttpserver.nim

```diff
@@ -271,21 +271,8 @@ proc processRequest(
         await client.sendStatus("417 Expectation Failed")
 
   # Read the body
-  # - Check for Content-length header
-  if request.headers.hasKey("Content-Length"):
-    var contentLength = 0
-    if parseSaturatedNatural(request.headers["Content-Length"], contentLength) == 0:
-      await request.respond(Http400, "Bad Request. Invalid Content-Length.")
-      return true
-    else:
-      if contentLength > server.maxBody:
-        await request.respondError(Http413)
-        return false
-      request.body = await client.recv(contentLength)
-      if request.body.len != contentLength:
-        await request.respond(Http400, "Bad Request. Content-Length does not match actual.")
-        return true
-  elif hasChunkedEncoding(request):
+  # - Check for Transfer-Encoding header
+  if hasChunkedEncoding(request):
```

Comment:
No one remarked on this in the PR yet, so I wanted to highlight that this reorders the check for the `Transfer-Encoding` header to be before the `Content-Length` header. Both headers should not appear on the same request according to the spec, so this should be fine, but I'd appreciate an explicit approval for this.

Reply:
Copied from https://github.com/nim-lang/Nim/pull/18052#issuecomment-850014687

```
The Content-Length header gets added to all requests sent by the client very deeply in the code, and the server does a check for this header before checking for the Transfer-Encoding. But since I couldn't get the client to stop adding the Content-Length header, my previous tests in this commit chain simply had both headers. This meant that the simplest way to get the right behaviour was moving the check for Transfer-Encoding to be before that of the Content-Length header. However, now that the test uses sockets (like the original code), there are no issues with both headers being present because I have complete control over what gets sent down the wire.
```